temporian/implementation/numpy/operators/filter_event.py

```diff
@@ -0,0 +1,57 @@
+from typing import Dict
+
+from temporian.core.operators.filter import FilterOperator
+from temporian.implementation.numpy.data.event import NumpyEvent
+from temporian.implementation.numpy.data.feature import NumpyFeature
+from temporian.implementation.numpy.data.sampling import NumpySampling
+from temporian.implementation.numpy import implementation_lib
+from temporian.implementation.numpy.operators.base import OperatorImplementation
+
+
+class FilterNumpyImplementation(OperatorImplementation):
+    """Filter timestamps from an event based on a condition."""
+
+    def __init__(self, operator: FilterOperator) -> None:
+        super().__init__(operator)
+
+    def __call__(
+        self, event: NumpyEvent, condition: NumpyEvent
+    ) -> Dict[str, NumpyEvent]:
+        new_sampling = {}
+        event_filtered_data = {}
+
+        for index, features in condition.data.items():
+            # get boolean mask from condition
+            condition_feature = features[0].data
+
+            # filtered sampling
+            filtered_sampling = event.sampling.data[index][condition_feature]
+
+            # if filtered sampling is empty, skip
```

Comment:
This is a good point. Modeling-wise, do you think returning an empty sampling would be better or worst?

Reply:
If I return an empty sampling we cannot have the following functionality:

```python
event_df = pd.DataFrame(
            [
                [1.0, 10.0, "A"],
                [2.0, np.nan, "A"],
                [3.0, 12.0, "B"],
                [4.0, 13.0, "B"],
                [5.0, 14.0, "C"],
                [6.0, 15.0, "C"],
            ],
            columns=["timestamp", "sales", "product"],
        )

condition_df = pd.DataFrame(
            [
                [1.0, True, "A"],
                [2.0, True, "A"],
                [3.0, True, "B"],
                [4.0, False, "B"],
                [5.0, False, "C"],
                [6.0, False, "C"],
            ],
            columns=["timestamp", "low_sales", "product"],
        )

expected_df = pd.DataFrame(
            [
                ["A", 1.0, 10.0],
                ["A", 2.0, np.nan],
                ["B", 3.0, 12.0],
            ],
            columns=[
                "product",
                "timestamp",
                "sales_filtered_by_low_sales",
            ],
        )
```

We would probably have to add rows for each sampling in the DF and not add any values on the features. I think it makes it a bit more counterintuitive.

Reply:
I prefer the alternative of removing the index level (what is currently implemented).